dataframely/collection.py

```diff
@@ -225,6 +226,58 @@ def sample(
         # 3) Eventually, we initialize the final collection and return
         return cls.validate(members)
 
+    @classmethod
+    def matches(cls, other: type["Collection"]) -> bool:
+        """Check whether this collection semantically matches another.
+
+        Args:
+            other: The collection to compare with.
+
+        Returns:
+            Whether the two collections are semantically equal.
+
+        Attention:
+            For custom filters, reliable comparison results are only guaranteed
+            if the filter always returns a static polars expression.
+            Otherwise, this function may falsely indicate a match.
+        """
+        schemas_lhs = cls.member_schemas()
+        schemas_rhs = other.member_schemas()
+
+        # Member names must match
+        if schemas_lhs.keys() != schemas_rhs.keys():
+            return False
+
+        # Member schemas must match
+        if not all(
+            schemas_lhs[name].matches(schemas_rhs[name]) for name in schemas_lhs
+        ):
+            return False
+
+        # Filter names must match
+        filters_lhs = cls._filters()
+        filters_rhs = other._filters()
+        if filters_lhs.keys() != filters_rhs.keys():
+            return False
+
+        # Computational graph of filter logic must match
+        # Evaluate on empty dataframes
+        empty_left = cls.create_empty()
+        empty_right = other.create_empty()
+
+        def filter_logic_matches(filter_left: Filter, filter_right: Filter) -> bool:
+            lf_left = filter_left.logic(empty_left)
+            lf_right = filter_right.logic(empty_right)
+            return lf_left.explain() == lf_right.explain()
```

Comment:
Can we use https://docs.pola.rs/api/python/stable/reference/lazyframe/api/polars.LazyFrame.serialize.html? `explain` runs optimization so it is considerably more expensive. `serialize` also seems to be more structured.